File: FirstTask/JavaScript/newton_sqrt/newton_sqrt.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

rl.question('Give a number, that you want to know the square root for: ', number => {
    let res = newton_sqrt(number);
    console.log('The square root of your number is (approximatly) ' + res);
    rl.close();
})

function newton_sqrt(number){
    let first = number;
    let latest = number;
    i = 1;
    while (i < 1000 && (latest * latest - first) > 0.0001) {
        latest = 0.5 * (latest + first / latest);
        i++;
    }
    return latest;
}

File: FirstTask/Haskell/newton_sqrt/newton_sqrt.hs
iteration x y = if (x*x-y) < 0.0001 then x else iteration (0.5 * (x + y/x)) y

main = do
    putStrLn "What number do you want to calculate the square root from?"
    input <- getLine
    let number = read input :: Double
    let number1 = iteration number number
    putStrLn ("Your square root is (approximatly): ")
    print number1